src/main/java/com/webfleet/oauth/service/feign/ResponsePayloadJackson2Deserializer.java

```diff
@@ -0,0 +1,116 @@
+package com.webfleet.oauth.service.feign;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.webfleet.oauth.service.feign.ResponsePayload;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+
+public final class ResponsePayloadJackson2Deserializer extends StdDeserializer<ResponsePayload>
+{
+
+    public ResponsePayloadJackson2Deserializer()
+    {
+        super(ResponsePayload.class);
+    }
+
+    @Override
+    public ResponsePayload deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException
+    {
+
+        String tokenValue = null;
+        String tokenType = null;
+        String refreshToken = null;
+        Long expiresIn = null;
+        Set<String> scope = null;
+        Map<String, Object> additionalInformation = new LinkedHashMap<>();
+
+        // TODO What should occur if a parameter exists twice
+        while (jp.nextToken() != JsonToken.END_OBJECT)
+        {
+            String name = jp.getCurrentName();
+            jp.nextToken();
+            if (ResponsePayload.ACCESS_TOKEN.equals(name))
+            {
+                tokenValue = jp.getText();
+            } else if (ResponsePayload.TOKEN_TYPE.equals(name))
```

Comment:
`else` should go to the next line if you follow WF style. Otherwise the `{` should go to the same line as `if`.